src/lib.rs

```diff
@@ -842,6 +843,15 @@ impl DirEntry {
     }
 }
 
+#[cfg(unix)]
+impl std::os::unix::fs::DirEntryExt for DirEntry {
```

Comment:
You'd think this is a perfect reason to implement `DirEntryExt` in the standard library but we shouldn't do that :( Those traits are going away at some point, so we need to define our own `DirEntryExt` with an `ino` method and implement that.

It'll be easier to put this trait and its implementation for `DirEntry` in a separate `unix.rs` file.